pandas/io/excel/_base.py

```diff
@@ -1616,31 +1615,13 @@ def __init__(
             if engine == "auto":
                 engine = get_default_engine(ext, mode="reader")
 
-        if engine == "xlrd" and xlrd_version is not None:
-            if ext is None:
-                # Need ext to determine ext in order to raise/warn
-                if isinstance(path_or_buffer, xlrd.Book):
-                    ext = "xls"
-                else:
-                    ext = inspect_excel_format(
-                        path_or_buffer, storage_options=storage_options
-                    )
-
-            # Pass through if ext is None, otherwise check if ext valid for xlrd
-            if ext and ext != "xls" and xlrd_version >= Version("2"):
-                raise ValueError(
-                    f"Your version of xlrd is {xlrd_version}. In xlrd >= 2.0, "
-                    f"only the xls format is supported. Install openpyxl instead."
-                )
-            elif ext and ext != "xls":
-                stacklevel = find_stack_level()
-                warnings.warn(
-                    f"Your version of xlrd is {xlrd_version}. In xlrd >= 2.0, "
-                    f"only the xls format is supported. Install "
-                    f"openpyxl instead.",
-                    FutureWarning,
-                    stacklevel=stacklevel,
-                )
+        if engine == "xlrd":
+            warnings.warn(
+                "The xlrd engine is no longer maintained and "
```

Comment:
So I may have been wrong prior; From this post,  https://groups.google.com/g/python-excel/c/IRa8IWq_4zk/m/Af8-hrRnAgAJ, I think technically `xlrd` will still be maintained just for `xls` files (though still appears highly discouraged)

Reply:
So I am not sure if this warning is entirely appropriate; maybe pandas can still "help" by raising if  engine is xlrd and the extension is not `xls`?

Reply:
@mroeschke - agree with your interpretation there, but then there is also https://github.com/pandas-dev/pandas/issues/28547#issuecomment-735437456. xlrd hasn't seen a commit since 2.0.1 was released; I think it's no longer maintained.

> So I am not sure if this warning is entirely appropriate; maybe pandas can still "help" by raising if engine is xlrd and the extension is not `xls`?

xlrd will raise in such a case with a meaningful error message (see the added test), are you thinking we should have more details?

Reply:
@cjw296 - would you be able to provide any clarity on the current status of maintenance for xlrd? Also, thoughts on this PR would be most appreciated!

Reply:
> xlrd will raise in such a case with a meaningful error message (see the added test), are you thinking we should have more details?

Ah okay I missed that. Nevermind, I think we don't need to raise a separate error then

Reply:
@rhshadrach - given that pandas developers pretty much [ignore](https://github.com/pandas-dev/pandas/issues/26813) anything [published](https://stackoverflow.com/a/65266497/216229) about xlrd and have [rejected](https://github.com/pandas-dev/pandas/pull/38522) attempts I've made to contribute to this project, I'm surprised you've bothered asking.

xlrd does what it does, its CI runs. If folks specifically need to parse xls files, it's likely their best option in Python.

Reply:
Sorry for the off-and-on responsiveness over the years @cjw296.

Given xlrd's status, I would slightly be against the warning given wild xls files probably still floating round and xlrd being the only mechanism to read them (in pandas).  WDYT @rhshadrach?

Reply:
> given wild xls files probably still floating round and xlrd being the only mechanism to read them (in pandas)

I agree with this part. The warning is more to tell users that xlrd won't likely work indefinitely. However, looking at the build, it doesn't use any third party libraries which makes me think it will work for quite some time until there is a breaking change in Python itself that impacts it. If we see an upcoming Python release that isn't compatible with xlrd and the community doesn't fix it we can warn then.

I'll remove the warning.